src/application.rs

```diff
@@ -38,7 +41,8 @@ pub struct Application {
     ui: Ui,
     // this is our model :)
     model: Rc<Model>,
-    raw_model: RawModel,
+    // pending requests
+    pending_requests: HashMap<u64, Rc<dyn Fn(&mut Application)>>,
```

Comment:
Rc, because you're not sharing this via threads, right?

Reply:
because `dyn` object doesn't have size. `Rc` does. you need something to wrap your closure into. Since we do not pass across threads `Rc` is enough